### ISSUE TITLE
pool.close does not wait for asynchronous resources to be destroyed

### ISSUE THREAD

**Issue by cressie176** [panates/lightning-pool#15]
Hello again. Thank you for fixing my previous [issue](https://github.com/panates/lightning-pool/issues/14) so quickly and sorry for raising another issue so soon.

This time I've found that `pool.close` does not wait for resources to be destroyed before yielding.

```js
const { Pool } = require('lightning-pool');

let i = 1;

const factory = {
  create: async function(opts) {
    console.log('CREATING', i);
    return new Promise((resolve) => {
      setTimeout(() => {
        resolve(i);
        console.log('CREATED', i);
        i++;
      }, 500)
    });
  },
  destroy: async function(resource) {
    console.log('DESTROYING', resource);
    return new Promise((resolve) => {
      setTimeout(() => {
        console.log('DESTROYED', resource);
        resolve(i);
      }, 500)
    })
  }
};

const pool = new Pool(factory, { max: 20, min: 10});

pool.start();

process.on('SIGINT', () => {
  pool.close(() => {
    console.log('CLOSED');
    process.exit(0);
  });
})

setInterval(() => {}, 60000)
```

### Output
```
CREATING 1
CREATING 1
CREATING 1
CREATING 1
CREATING 1
CREATING 1
CREATING 1
CREATING 1
CREATING 1
CREATING 1
CREATED 1
CREATED 2
CREATED 3
CREATED 4
CREATED 5
CREATED 6
CREATED 7
CREATED 8
CREATED 9
CREATED 10
^CDESTROYING 1
DESTROYING 2
DESTROYING 3
DESTROYING 4
DESTROYING 5
DESTROYING 6
DESTROYING 7
DESTROYING 8
DESTROYING 9
DESTROYING 10
CLOSED
```
The above output has no `DESTROYED` output

**Comment by erayhanoglu** [comment]
Released v4.10.0 with this fix. Thank you for reporting 👍

**Comment by cressie176** [comment]
Thanks again
